### ISSUE TITLE
Segmentation fault happens when using index_add

### ISSUE THREAD

**Issue by QuantumCoder4** [pytorch/pytorch#122117]
### 🐛 Describe the bug

When I sent the following to both torch.Tensor.index_add and torch.index_add, a segmentation fault happens

```
import numpy as np
np.random.seed(1234)
import torch

input = torch.tensor(4, dtype=torch.float32)
dim = 0
index = torch.tensor(19, dtype=torch.int64)
tensor2 = torch.tensor(np.random.randint(0, 50, (1,3,4,4)), dtype=torch.float32)
```

### Versions

Versions of relevant libraries:
[pip3] numpy==1.26.2
[pip3] torch==2.2.1
[pip3] triton==2.2.0
[conda] numpy                     1.26.2                   pypi_0    pypi
[conda] torch                     2.2.1                    pypi_0    pypi
[conda] triton                    2.2.0                    pypi_0    pypi

**Comment by sanchitintel** [comment]
Not reproducible with the main branch, as it was fixed in #114929.
You'd encounter this runtime error instead:
```
RuntimeError: source tensor shape must match self tensor shape, excluding the specified dimension. Got self.shape = [] source.shape = [1, 3, 4, 4]
```

From https://pytorch.org/docs/stable/generated/torch.Tensor.index_add_.html#torch.Tensor.index_add_,
<img width="766" alt="image" src="https://github.com/pytorch/pytorch/assets/90872293/a4787012-0265-4ad5-9ee9-1ea130ec690f">

**Comment by QuantumCoder4** [comment]
Thank for fixing it! Let me close this issue.
